pkg/prometheus/promcfg.go

```diff
@@ -470,14 +480,16 @@ func (cg *ConfigGenerator) Generate(
 
 	cfg := yaml.MapSlice{}
 
+	// TODO(slashpai): Remove this default assignment after v0.57 since this is set at CRD level
 	scrapeInterval := "30s"
 	if p.Spec.ScrapeInterval != "" {
-		scrapeInterval = p.Spec.ScrapeInterval
+		scrapeInterval = string(p.Spec.ScrapeInterval)
 	}
```

Comment:
this can be removed now since the returned object will have a default value of 30s already.

Reply:
We still have unit tests for this so I have added comment for this to remove after v0.57

Reply:
hmm I wish there was a way for the tests to get a Prometheus object with all defaults...